inc/datahandlers/user.php

```diff
@@ -888,6 +888,20 @@ function verify_checkfields()
 	}
 
 	/**
+	 * Verifies if the selected timezone is valid.
+	 *
+	 * @return boolean True when valid, false when invalid.
+	 */
+	function verify_timezone()
+	{
+		$user = &$this->data;
+
+		$timezones = get_supported_timezones();
+
+		return array_key_exists($user['timezone'], $timezones);
```

Comment:
Why would you force the default value if it is wrong? It should return an error and displayed to the user accordingly.

Just comparing your code to the above checks tells me its inconsistency. The only case where this is true is when checking the 'username'.